pallets/pool-registry/src/lib.rs

```diff
@@ -238,6 +245,8 @@ pub mod pallet {
 			// admin as our depositor.
 			let depositor = ensure_signed(origin).unwrap_or(admin.clone());
 
+			Pools::<T>::set(pool_id, Some(PoolRegistrationStatus::Registered));
```

Comment:
We should probably have a check that if the pool is already registered, we error out with `PoolIdAlreadyRegistered`?

Reply:
Good point! Updated!